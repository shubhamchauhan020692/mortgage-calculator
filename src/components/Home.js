import React, { useState } from "react";
import { Charts } from "./Charts";
import { SummaryTable } from "./SummaryTable";
import { validateInputs } from "../utilities/validate";

import { calculateMortage } from "../utilities/calculations";
import { MortgageForm } from "./MortgageForm";
import { Header } from "./Header";

export const Home = () => {
  const [mortgageAmt, setMortgageAmt] = useState(50000);
  const [rate, setRate] = useState(5);
  const [amortizationPeriod, setAmortizationPeriod] = useState(25);
  const [paymentFrequency, setPaymentFrequency] = useState(12);
  const [term, setTerm] = useState(5);
  const [summaryTableData, setSummaryTableData] = useState(5);
  const [graphData, setGraphData] = useState(null);
  const [barData, setBarData] = useState({});
  const [errors, setErrors] = useState({
    mortgageAmt: "",
    rate: "",
    amortizationPeriod: "",
    paymentFrequency: "",
    term: "",
  });

  const handleInputChange = (e) => {
    setErrors({
      ...errors,
      [e.target.name]: "",
    });
    switch (e.target.name) {
      case "mortgageAmt": {
        setMortgageAmt(e.target.value);
        break;
      }
      case "rate": {
        setRate(e.target.value);
        break;
      }
      case "amortizationPeriod": {
        setAmortizationPeriod(e.target.value);
        break;
      }
      case "paymentFrequency": {
        setPaymentFrequency(e.target.value);
        break;
      }
      case "term": {
        setTerm(e.target.value);
        break;
      }
      default:
        return;
    }
  };

  const handleInputBlur = () => {
    const formErrors = validateInputs(
      mortgageAmt,
      rate,
      amortizationPeriod,
      term
    );
    formErrors &&
      setErrors({
        ...errors,
        ...formErrors,
      });

    /** TODO : If no error only then display */
    const { summaryTableData, graphData, barData } = calculateMortage(
      mortgageAmt,
      rate,
      amortizationPeriod,
      paymentFrequency,
      term
    );
    setSummaryTableData(summaryTableData);
    setGraphData(graphData);
    setBarData(barData);
  };

  console.log("errors", errors);
  return (
    <div className="main-container">
      <MortgageForm
        mortgageAmt={mortgageAmt}
        rate={rate}
        amortizationPeriod={amortizationPeriod}
        paymentFrequency={paymentFrequency}
        term={term}
        handleInputChange={handleInputChange}
        handleInputBlur={handleInputBlur}
        errors={errors}
      />
      <SummaryTable summaryTableData={summaryTableData} />
      {graphData && <Charts graphData={graphData} barData={barData} />}
    </div>
  );
};
