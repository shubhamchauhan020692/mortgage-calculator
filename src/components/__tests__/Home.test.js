import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Home } from '../Home';

describe('Home', () => {
  afterEach(cleanup);

  it('renders correctly', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
