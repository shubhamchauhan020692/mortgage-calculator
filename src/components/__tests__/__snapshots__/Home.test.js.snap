// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<DocumentFragment>
  <div
    class="jumbotron col-12 col-sm-6"
    data-testid="mortgage-form-container"
    id="mortgage-form-container"
  >
    <div
      class="row align-items-center mb-3"
    >
      <div
        class="col-12 col-sm-6"
      >
        <label
          for="basic-url"
        >
          Mortgage Amount
        </label>
        <a
          class="tooltip-help"
          data-tip="The amount you expect to borrow from your financial institution."
          href="/#"
        >
          ?
        </a>
      </div>
      <div
        class="col-12 col-sm-6"
      >
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              $
            </span>
          </div>
          <input
            aria-label="Amount (to the nearest dollar)"
            class="form-control"
            data-testid="mortgageAmt"
            id="mortgageAmt"
            name="mortgageAmt"
            placeholder="Value nearest to dollar amount"
            type="number"
            value="50000"
          />
          <div
            class="input-group-append"
          >
            <span
              class="input-group-text"
            >
              .00
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row align-items-center mb-3 align-items-center mb-3"
    >
      <div
        class="col-12 col-sm-6"
      >
        <label
          for="interestRate"
        >
          Interest Rate
        </label>
        <a
          class="tooltip-help"
          data-tip="Annual interest rate for this mortgage."
          href="/#"
        >
          ?
        </a>
      </div>
      <div
        class="col-12 col-sm-6"
      >
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              %
            </span>
          </div>
          <input
            aria-label="Interest Rate"
            class="form-control"
            data-testid="rate"
            id="rate"
            max="100"
            min="0"
            name="rate"
            placeholder="Annual Interest Rate"
            type="number"
            value="5"
          />
        </div>
      </div>
    </div>
    <div
      class="row align-items-center mb-3"
    >
      <div
        class="col-12 col-sm-6"
      >
        <label
          for="amortizationPeriod"
        >
          Amortization Period
        </label>
        <a
          class="tooltip-help"
          data-tip="The number of years and months over which you will repay this loan. The most common amortization period is 25 years."
          href="/#"
        >
          ?
        </a>
      </div>
      <div
        class="col-12 col-sm-6"
      >
        <input
          aria-label="Amortization Period"
          class="form-control"
          data-testid="amortizationPeriod"
          id="amortizationPeriod"
          max="30"
          min="1"
          name="amortizationPeriod"
          placeholder="Years"
          type="number"
          value="25"
        />
      </div>
    </div>
    <div
      class="row align-items-center mb-3"
    >
      <div
        class="col-12 col-sm-6"
      >
        <label
          for="paymentFrequency"
        >
          Payment Frequency
        </label>
        <a
          class="tooltip-help"
          data-tip="By choosing an accelerated payment frequency, you can reduce your amortization period and save thousands of dollars in interest in the long run."
          href="/#"
        >
          ?
        </a>
      </div>
      <div
        class="col-12 col-sm-6"
      >
        <select
          data-testid="paymentFrequency"
          id="paymentFrequency"
          name="paymentFrequency"
        >
          <option
            id="0"
            value="52"
          >
            Weekly
          </option>
          <option
            id="1"
            value="26"
          >
            Bi-Weekly (every 2 weeks)
          </option>
          <option
            id="2"
            value="12"
          >
            Monthly (12x per year)
          </option>
          <option
            id="3"
            value="24"
          >
            Semi-monthly (24x per year)
          </option>
        </select>
      </div>
    </div>
    <div
      class="row align-items-center mb-3"
    >
      <div
        class="col-12 col-sm-6"
      >
        <label
          for="term"
        >
          Term
        </label>
        <a
          class="tooltip-help"
          data-tip="The number of term years."
          href="/#"
        >
          ?
        </a>
      </div>
      <div
        class="col-12 col-sm-6"
      >
        <select
          data-testid="term"
          id="term"
          name="term"
        >
          <option
            id="0"
            value="1"
          >
            1
          </option>
          <option
            id="1"
            value="2"
          >
            2
          </option>
          <option
            id="2"
            value="3"
          >
            3
          </option>
          <option
            id="3"
            value="4"
          >
            4
          </option>
          <option
            id="4"
            value="5"
          >
            5
          </option>
          <option
            id="5"
            value="6"
          >
            6
          </option>
          <option
            id="6"
            value="7"
          >
            7
          </option>
          <option
            id="7"
            value="8"
          >
            8
          </option>
          <option
            id="8"
            value="9"
          >
            9
          </option>
          <option
            id="9"
            value="10"
          >
            10
          </option>
          <option
            id="10"
            value="11"
          >
            11
          </option>
          <option
            id="11"
            value="12"
          >
            12
          </option>
          <option
            id="12"
            value="13"
          >
            13
          </option>
          <option
            id="13"
            value="14"
          >
            14
          </option>
          <option
            id="14"
            value="15"
          >
            15
          </option>
          <option
            id="15"
            value="16"
          >
            16
          </option>
          <option
            id="16"
            value="17"
          >
            17
          </option>
          <option
            id="17"
            value="18"
          >
            18
          </option>
          <option
            id="18"
            value="19"
          >
            19
          </option>
          <option
            id="19"
            value="20"
          >
            20
          </option>
          <option
            id="20"
            value="21"
          >
            21
          </option>
          <option
            id="21"
            value="22"
          >
            22
          </option>
          <option
            id="22"
            value="23"
          >
            23
          </option>
          <option
            id="23"
            value="24"
          >
            24
          </option>
          <option
            id="24"
            value="25"
          >
            25
          </option>
          <option
            id="25"
            value="26"
          >
            26
          </option>
          <option
            id="26"
            value="27"
          >
            27
          </option>
          <option
            id="27"
            value="28"
          >
            28
          </option>
          <option
            id="28"
            value="29"
          >
            29
          </option>
          <option
            id="29"
            value="30"
          >
            30
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    class="recharts-wrapper"
    style="position: relative; cursor: default; width: 730px; height: 250px;"
  >
    <svg
      class="recharts-surface"
      height="250"
      version="1.1"
      viewBox="0 0 730 250"
      width="730"
    >
      <defs>
        <clippath
          id="recharts2-clip"
        >
          <rect
            height="210"
            width="640"
            x="60"
            y="10"
          />
        </clippath>
      </defs>
      <defs>
        <lineargradient
          id="colorUv"
          x1="0"
          x2="0"
          y1="0"
          y2="1"
        >
          <stop
            offset="5%"
            stop-color="#8884d8"
            stop-opacity="0.8"
          />
          <stop
            offset="95%"
            stop-color="#8884d8"
            stop-opacity="0"
          />
        </lineargradient>
      </defs>
      <g
        class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
      >
        <line
          class="recharts-cartesian-axis-line"
          fill="none"
          height="30"
          stroke="#666"
          width="640"
          x="60"
          x1="60"
          x2="700"
          y="220"
          y1="220"
          y2="220"
        />
        <g
          class="recharts-cartesian-axis-ticks"
        >
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              class="recharts-cartesian-axis-tick-line"
              fill="none"
              height="30"
              stroke="#666"
              width="640"
              x="60"
              x1="60"
              x2="60"
              y="220"
              y1="226"
              y2="220"
            />
            <text
              class="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height="30"
              stroke="none"
              text-anchor="middle"
              width="640"
              x="60"
              y="228"
            >
              <tspan
                dy="0.71em"
                x="60"
              >
                0
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              class="recharts-cartesian-axis-tick-line"
              fill="none"
              height="30"
              stroke="#666"
              width="640"
              x="60"
              x1="166.66666666666669"
              x2="166.66666666666669"
              y="220"
              y1="226"
              y2="220"
            />
            <text
              class="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height="30"
              stroke="none"
              text-anchor="middle"
              width="640"
              x="166.66666666666669"
              y="228"
            >
              <tspan
                dy="0.71em"
                x="166.66666666666669"
              >
                100
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              class="recharts-cartesian-axis-tick-line"
              fill="none"
              height="30"
              stroke="#666"
              width="640"
              x="60"
              x1="273.33333333333337"
              x2="273.33333333333337"
              y="220"
              y1="226"
              y2="220"
            />
            <text
              class="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height="30"
              stroke="none"
              text-anchor="middle"
              width="640"
              x="273.33333333333337"
              y="228"
            >
              <tspan
                dy="0.71em"
                x="273.33333333333337"
              >
                200
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              class="recharts-cartesian-axis-tick-line"
              fill="none"
              height="30"
              stroke="#666"
              width="640"
              x="60"
              x1="380"
              x2="380"
              y="220"
              y1="226"
              y2="220"
            />
            <text
              class="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height="30"
              stroke="none"
              text-anchor="middle"
              width="640"
              x="380"
              y="228"
            >
              <tspan
                dy="0.71em"
                x="380"
              >
                300
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              class="recharts-cartesian-axis-tick-line"
              fill="none"
              height="30"
              stroke="#666"
              width="640"
              x="60"
              x1="486.6666666666667"
              x2="486.6666666666667"
              y="220"
              y1="226"
              y2="220"
            />
            <text
              class="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height="30"
              stroke="none"
              text-anchor="middle"
              width="640"
              x="486.6666666666667"
              y="228"
            >
              <tspan
                dy="0.71em"
                x="486.6666666666667"
              >
                400
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              class="recharts-cartesian-axis-tick-line"
              fill="none"
              height="30"
              stroke="#666"
              width="640"
              x="60"
              x1="593.3333333333334"
              x2="593.3333333333334"
              y="220"
              y1="226"
              y2="220"
            />
            <text
              class="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height="30"
              stroke="none"
              text-anchor="middle"
              width="640"
              x="593.3333333333334"
              y="228"
            >
              <tspan
                dy="0.71em"
                x="593.3333333333334"
              >
                500
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              class="recharts-cartesian-axis-tick-line"
              fill="none"
              height="30"
              stroke="#666"
              width="640"
              x="60"
              x1="700"
              x2="700"
              y="220"
              y1="226"
              y2="220"
            />
            <text
              class="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height="30"
              stroke="none"
              text-anchor="middle"
              width="640"
              x="700"
              y="228"
            >
              <tspan
                dy="0.71em"
                x="700"
              >
                600
              </tspan>
            </text>
          </g>
        </g>
      </g>
      <g
        class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
      >
        <line
          class="recharts-cartesian-axis-line"
          fill="none"
          height="210"
          stroke="#666"
          width="60"
          x="0"
          x1="60"
          x2="60"
          y="10"
          y1="10"
          y2="220"
        />
        <g
          class="recharts-cartesian-axis-ticks"
        >
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              class="recharts-cartesian-axis-tick-line"
              fill="none"
              height="210"
              stroke="#666"
              width="60"
              x="0"
              x1="54"
              x2="60"
              y="10"
              y1="220"
              y2="220"
            />
            <text
              class="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height="210"
              stroke="none"
              text-anchor="end"
              width="60"
              x="52"
              y="220"
            >
              <tspan
                dy="0.355em"
                x="52"
              >
                -100000
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              class="recharts-cartesian-axis-tick-line"
              fill="none"
              height="210"
              stroke="#666"
              width="60"
              x="0"
              x1="54"
              x2="60"
              y="10"
              y1="167.5"
              y2="167.5"
            />
            <text
              class="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height="210"
              stroke="none"
              text-anchor="end"
              width="60"
              x="52"
              y="167.5"
            >
              <tspan
                dy="0.355em"
                x="52"
              >
                -50000
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              class="recharts-cartesian-axis-tick-line"
              fill="none"
              height="210"
              stroke="#666"
              width="60"
              x="0"
              x1="54"
              x2="60"
              y="10"
              y1="115"
              y2="115"
            />
            <text
              class="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height="210"
              stroke="none"
              text-anchor="end"
              width="60"
              x="52"
              y="115"
            >
              <tspan
                dy="0.355em"
                x="52"
              >
                0
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              class="recharts-cartesian-axis-tick-line"
              fill="none"
              height="210"
              stroke="#666"
              width="60"
              x="0"
              x1="54"
              x2="60"
              y="10"
              y1="62.5"
              y2="62.5"
            />
            <text
              class="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height="210"
              stroke="none"
              text-anchor="end"
              width="60"
              x="52"
              y="62.5"
            >
              <tspan
                dy="0.355em"
                x="52"
              >
                50000
              </tspan>
            </text>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              class="recharts-cartesian-axis-tick-line"
              fill="none"
              height="210"
              stroke="#666"
              width="60"
              x="0"
              x1="54"
              x2="60"
              y="10"
              y1="10"
              y2="10"
            />
            <text
              class="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height="210"
              stroke="none"
              text-anchor="end"
              width="60"
              x="52"
              y="10"
            >
              <tspan
                dy="0.355em"
                x="52"
              >
                100000
              </tspan>
            </text>
          </g>
        </g>
      </g>
      <g
        class="recharts-cartesian-grid"
      >
        <g
          class="recharts-cartesian-grid-horizontal"
        >
          <line
            fill="none"
            height="210"
            stroke="#ccc"
            stroke-dasharray="3 3"
            width="640"
            x="60"
            x1="60"
            x2="700"
            y="10"
            y1="220"
            y2="220"
          />
          <line
            fill="none"
            height="210"
            stroke="#ccc"
            stroke-dasharray="3 3"
            width="640"
            x="60"
            x1="60"
            x2="700"
            y="10"
            y1="167.5"
            y2="167.5"
          />
          <line
            fill="none"
            height="210"
            stroke="#ccc"
            stroke-dasharray="3 3"
            width="640"
            x="60"
            x1="60"
            x2="700"
            y="10"
            y1="115"
            y2="115"
          />
          <line
            fill="none"
            height="210"
            stroke="#ccc"
            stroke-dasharray="3 3"
            width="640"
            x="60"
            x1="60"
            x2="700"
            y="10"
            y1="62.5"
            y2="62.5"
          />
          <line
            fill="none"
            height="210"
            stroke="#ccc"
            stroke-dasharray="3 3"
            width="640"
            x="60"
            x1="60"
            x2="700"
            y="10"
            y1="10"
            y2="10"
          />
        </g>
        <g
          class="recharts-cartesian-grid-vertical"
        >
          <line
            fill="none"
            height="210"
            stroke="#ccc"
            stroke-dasharray="3 3"
            width="640"
            x="60"
            x1="60"
            x2="60"
            y="10"
            y1="10"
            y2="220"
          />
          <line
            fill="none"
            height="210"
            stroke="#ccc"
            stroke-dasharray="3 3"
            width="640"
            x="60"
            x1="166.66666666666669"
            x2="166.66666666666669"
            y="10"
            y1="10"
            y2="220"
          />
          <line
            fill="none"
            height="210"
            stroke="#ccc"
            stroke-dasharray="3 3"
            width="640"
            x="60"
            x1="273.33333333333337"
            x2="273.33333333333337"
            y="10"
            y1="10"
            y2="220"
          />
          <line
            fill="none"
            height="210"
            stroke="#ccc"
            stroke-dasharray="3 3"
            width="640"
            x="60"
            x1="380"
            x2="380"
            y="10"
            y1="10"
            y2="220"
          />
          <line
            fill="none"
            height="210"
            stroke="#ccc"
            stroke-dasharray="3 3"
            width="640"
            x="60"
            x1="486.6666666666667"
            x2="486.6666666666667"
            y="10"
            y1="10"
            y2="220"
          />
          <line
            fill="none"
            height="210"
            stroke="#ccc"
            stroke-dasharray="3 3"
            width="640"
            x="60"
            x1="593.3333333333334"
            x2="593.3333333333334"
            y="10"
            y1="10"
            y2="220"
          />
          <line
            fill="none"
            height="210"
            stroke="#ccc"
            stroke-dasharray="3 3"
            width="640"
            x="60"
            x1="700"
            x2="700"
            y="10"
            y1="10"
            y2="220"
          />
        </g>
      </g>
      <g
        class="recharts-layer recharts-area"
      >
        <g
          class="recharts-layer"
        >
          <defs>
            <clippath
              id="animationClipPath-recharts-area-3"
            >
              <rect
                height="212"
                width="0"
                x="60"
                y="0"
              />
            </clippath>
          </defs>
          <g
            class="recharts-layer"
            clip-path="url(#animationClipPath-recharts-area-3)"
          >
            <g
              class="recharts-layer"
            >
              <path
                class="recharts-curve recharts-area-area"
                d="M60,10C95.55555555555557,21.190550000000005,131.11111111111111,32.3811,166.66666666666669,43.571650000000005C202.22222222222226,54.76220000000001,237.7777777777778,65.95275,273.33333333333337,77.1433C308.8888888888889,88.33384999999998,344.44444444444446,99.52439999999999,380,110.71494999999999C415.55555555555554,121.90549999999999,451.11111111111114,133.09605,486.6666666666667,144.2866C522.2222222222223,155.47715,557.7777777777778,166.66770000000002,593.3333333333334,177.85825000000003C628.8888888888889,189.04880000000003,664.4444444444445,200.23935,700,211.42989999999998L700,115C664.4444444444445,115,628.8888888888889,115,593.3333333333334,115C557.7777777777778,115,522.2222222222223,115,486.6666666666667,115C451.11111111111114,115,415.55555555555554,115,380,115C344.44444444444446,115,308.8888888888889,115,273.33333333333337,115C237.7777777777778,115,202.22222222222226,115,166.66666666666669,115C131.11111111111111,115,95.55555555555557,115,60,115Z"
                fill="url(#colorUv)"
                fill-opacity="1"
                height="210"
                stroke="none"
                width="640"
              />
              <path
                class="recharts-curve recharts-area-curve"
                d="M60,10C95.55555555555557,21.190550000000005,131.11111111111111,32.3811,166.66666666666669,43.571650000000005C202.22222222222226,54.76220000000001,237.7777777777778,65.95275,273.33333333333337,77.1433C308.8888888888889,88.33384999999998,344.44444444444446,99.52439999999999,380,110.71494999999999C415.55555555555554,121.90549999999999,451.11111111111114,133.09605,486.6666666666667,144.2866C522.2222222222223,155.47715,557.7777777777778,166.66770000000002,593.3333333333334,177.85825000000003C628.8888888888889,189.04880000000003,664.4444444444445,200.23935,700,211.42989999999998"
                fill="none"
                fill-opacity="1"
                height="210"
                stroke="#8884d8"
                width="640"
              />
            </g>
          </g>
        </g>
      </g>
    </svg>
    <div
      class="recharts-tooltip-wrapper"
      style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px;"
    >
      <div
        class="recharts-default-tooltip"
        style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
      >
        <p
          class="recharts-tooltip-label"
          style="margin: 0px;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
